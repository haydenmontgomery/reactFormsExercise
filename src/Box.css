.Box {
  position: relative;
  border: 2px black;
  border-style: solid;
}