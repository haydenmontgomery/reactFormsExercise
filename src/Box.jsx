import React from "react";
import "./Box.css";

const Box = ({idx, width, height, bgcolor, deleteBox}) => {
  return (
    <div style={{
      position: "relative",
      marginBottom: "20px"
    }}>
      <button
        style={{
        position: "absolute",
        top: "10px",
        left: "0px"
        }}
      onClick={() => deleteBox(idx)}
      >
        Delete!
      </button>
    <div
      className="Box"
      style={{
        backgroundColor: bgcolor,
        width: `${width}px`,
        height: `${height}px`,
        position: "absoluute",
        left: "100px"
      }}
    >
    </div>
    </div>
  )

}

export default Box;