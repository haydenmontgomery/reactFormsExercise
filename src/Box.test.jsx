import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Box from "./Box";
import { it, expect } from "vitest";

it("renders without crashing", function () {
  render(<Box />);
});

it("matches snapshot", function () {
  const { asFragment } = render(<Box />);
  expect(asFragment()).toMatchSnapshot();
});