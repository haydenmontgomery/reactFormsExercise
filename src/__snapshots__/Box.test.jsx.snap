// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    style="position: relative; margin-bottom: 20px;"
  >
    <button
      style="position: absolute; top: 10px; left: 0px;"
    >
      Delete!
    </button>
    <div
      class="Box"
      style="left: 100px;"
    />
  </div>
</DocumentFragment>
`;
